dbos/_dbos_config.py

```diff
@@ -71,12 +173,53 @@ class ConfigFile(TypedDict, total=False):
     """
 
     name: str
-    language: str
     runtimeConfig: RuntimeConfig
     database: DatabaseConfig
     telemetry: Optional[TelemetryConfig]
     env: Dict[str, str]
-    application: Dict[str, Any]
+
+
+def is_config_file(obj: object) -> TypeGuard[ConfigFile]:
+    return (
+        isinstance(obj, dict)
+        and "name" in obj
+        and "runtimeConfig" in obj
+        and isinstance(obj.get("name"), str)
+        and isinstance(obj.get("runtimeConfig"), dict)
+    )
+
+
+def translate_dbos_config_to_config_file(config: DBOSConfig) -> ConfigFile:
+    db_config = parse_db_string_to_dbconfig(config["db_string"])
+    if "sys_db_name" in config:
+        db_config["sys_db_name"] = config.get("sys_db_name")
+
+    # Start with the mandatory fields
+    translated_config: ConfigFile = {
+        "name": config["name"],
+        "database": db_config,
+        "runtimeConfig": {
+            "start": [],
+        },
+    }
+    # Add admin_port to runtimeConfig if present
+    if "admin_port" in config:
+        translated_config["runtimeConfig"]["admin_port"] = config["admin_port"]
```

Comment:
This could still be `None`, but the admin server ctor has a default value     `def __init__(self, dbos: DBOS, port: int = 3001) -> None:`